FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/sockets_freertos.c

```diff
@@ -0,0 +1,162 @@
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* Standard includes. */
+#include <string.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_Sockets.h"
+
+#include "sockets_freertos.h"
+
+/*-----------------------------------------------------------*/
+
+/* Maximum number of times to call FreeRTOS_recv when initiating a graceful shutdown. */
+#ifndef TRANSPORT_FREERTOS_SHUTDOWN_LOOPS
+    #define TRANSPORT_FREERTOS_SHUTDOWN_LOOPS    ( 3 )
+#endif
+
+/* A negative error code indicating a network failure. */
+#define TRANSPORT_FREERTOS_NETWORK_ERROR    ( -1 )
+
+/*-----------------------------------------------------------*/
+
+BaseType_t Sockets_Connect( Socket_t * pTcpSocket,
+                            const char * pHostName,
+                            uint16_t port,
+                            uint32_t receiveTimeoutMs,
+                            uint32_t sendTimeoutMs )
+{
+    Socket_t tcpSocket = FREERTOS_INVALID_SOCKET;
+    BaseType_t socketStatus = 0;
+    struct freertos_sockaddr serverAddress = { 0 };
+    TickType_t transportTimeout = 0;
+
+    /* Create a new TCP socket. */
+    tcpSocket = FreeRTOS_socket( FREERTOS_AF_INET, FREERTOS_SOCK_STREAM, FREERTOS_IPPROTO_TCP );
+
+    if( tcpSocket == FREERTOS_INVALID_SOCKET )
+    {
+        LogError( ( "Failed to create new socket." ) );
+        socketStatus = TRANSPORT_FREERTOS_NETWORK_ERROR;
+    }
+    else
+    {
+        LogDebug( ( "Created new TCP socket." ) );
+
+        /* Connection parameters. */
+        serverAddress.sin_family = FREERTOS_AF_INET;
+        serverAddress.sin_port = FreeRTOS_htons( port );
+        serverAddress.sin_addr = FreeRTOS_gethostbyname( pHostName );
+        serverAddress.sin_len = ( uint8_t ) sizeof( serverAddress );
+
+        /* Check for errors from DNS lookup. */
+        if( serverAddress.sin_addr == 0 )
+        {
+            LogError( ( "Failed to resolve %s.", pHostName ) );
```

Comment:
(Nit) To follow the logging standard format of <Failure>:<Cause>:<Data>
```suggestion
            LogError( ( "Failed to connect to server: DNS resolution failed: Hostname=%s.", pHostName ) );
```